src/python/foglamp/translators/omf_translator.py

```diff
@@ -0,0 +1,732 @@
+    "i000001": "operation successfully completed",
+    "i000002": __module__ + " - Started",
+    "i000003": __module__ + " - Execution completed.",
+
+    # Warning / Error messages
+    "e000001": __module__ + " - generic error.",
+    "e000002": __module__ + " - cannot retrieve the starting point for sending operation - error details |{0}|.",
+    "e000003": __module__ + " - cannot update the reached position - error details |{0}|.",
+    "e000004": __module__ + " - cannot complete the sending operation - error details |{0}|.",
+    "e000005": __module__ + " - cannot configure the logging mechanism. - error details |{0}|.",
+    "e000006": __module__ + " - cannot initialize the plugin. - error details |{0}|.",
+    "e000007": __module__ + " - an error occurred during the OMF request. - error details |{0}|.",
+    "e000008": __module__ + " - an error occurred during the OMF's objects creation. - error details |{0}|.",
+    "e000009": __module__ + " - cannot retrieve information about the sensor.",
+    "e000010": __module__ + " - unable ro create the JSON message.",
+
+}
+"""Used messages"""
+
+# Logger
+_log       = ""
+_handler   = ""
+_formatter = ""
+
+# SA related
+_readings_values_tbl = sa.Table(
+    'readings',
+    sa.MetaData(),
+    sa.Column('id', sa.BigInteger, primary_key=True),
+    sa.Column('asset_code', sa.types.VARCHAR(50)),
+    sa.Column('read_key', sa.types.VARCHAR(50)),
+    sa.Column('user_ts', sa.types.TIMESTAMP),
+    sa.Column('reading', JSONB))
+"""Contains the information to send to OMF"""
+
+
+# PI Server references
+_server_name    = ""
+_relay_url      = ""
+_producer_token = ""
+
+# The size of a block of readings to send in each transmission.
+_block_size = 20
+
+# OMF's objects creation
+_types           = ""
+_sensor_id       = ""
+_measurement_id  = ""
+
+
+# OMF object's attributes
+_sensor_location = "S.F."
+
+# OMF's types definitions
+_type_id             = "8"
+_type_measurement_id = "omf_trans_type_measurement_" + _type_id
+_type_sensor_id      = "omf_trans_type_sensor_id_" + _type_id
+
+
+# DB operations
+_pg_conn = ""
+_pg_cur  = ""
+
+
+#
+# Functions
+#
+def plugin_initialize():
+    """Initializes the OMF plugin for the sending of blocks of readings to the PI Connector.
+
+    Retrieves the configuration for :
+        relay_url      - URL           - The URL of the PI Connector to send data to.
+        producer_token - producerToken - The producer token that represents this FogLAMP stream
+        types          - OMFTypes      - A JSON object that contains the OMF type definitions for this stream
+
+    Returns:
+        status: True if successful, False otherwise.
+
+    Raises:
+        Exception: Fails to initialise the plugin
+    """
+
+    global _server_name
+    global _relay_url
+    global _producer_token
+    global _types
+
+    global _sensor_location
+    global _sensor_id
+    global _measurement_id
+
+    global _type_measurement_id
+    global _type_sensor_id
+
```

Comment:
they are used by multiple functions and the code is still in evolution/changing so using as global, private to the module, it's pretty handy.